nomt/src/beatree/leaf/node.rs

```diff
@@ -112,6 +110,13 @@ impl LeafNode {
             )
         }
     }
+
+    // Extract a key from the given cell pointer
+    pub fn extract_key(cell_pointer: &[u8; 34]) -> Key {
```

Comment:
style: it's better to have the function as a free function in the module. Or rename to `Key::extract`.